code-challenges.test.js

```diff
@@ -22,9 +22,49 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("secretCodeWord", () => {
+    it("should return a coded message", () => {
+        expect(secretCodeWord(secretCodeWord1)).toBe("L4ck4d41s1c4l");
+        expect(secretCodeWord(secretCodeWord2)).toBe("G0bbl3dyg00k");
+        expect(secretCodeWord(secretCodeWord3)).toBe("3cc3ntr1c");
+    })
+})
+
 
 // b) Create the function that makes the test pass.
 
+// Pseudocode:
+// 1. Create a function that takes in a string and returns a coded message.
+// 2. Create a function that takes in an array and a letter and returns an array of words that contain the letter.
+// use for loop iteration to check if the letter is in the word
+// for (let i = 0; i < array.length; i++) let index 0, index array length the i++ of the whole array.
+//     if (array[i].includes(letter))
+//         codedWord += array[i] to a number string.
+//    else if (array[i].includes(letter))
+//        codedWord += array[i] to a number string.
+//  this repeats to check if the letter is in the word.
+//  the assignment operator keeps the value of the last iteration.
+// return codedWord
+
+
+const secretCodeWord = (word) => {
+    let codedWord = "";
+    for (let i = 0; i < word.length; i++) {
+        if (word[i] === "a") {
+            codedWord += "4";
+        } else if (word[i] === "e") {
+            codedWord += "3";
+        } else if (word[i] === "i") {
+            codedWord += "1";
+        } else if (word[i] === "o") {
+            codedWord += "0";
+        } else {
+            codedWord += word[i];
+        }
+    }
+    return codedWord;
+}
```

Comment:
Nice!  Your logic looks good here.  You may run into a failing test because of casing, but overall this looks great!